orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/tasks/providers/cf/CloudFoundryWaitForDeployServiceTask.java

```diff
@@ -31,6 +34,41 @@ public CloudFoundryWaitForDeployServiceTask(OortService oortService) {
     super(oortService);
   }
 
+  @Nonnull
+  @Override
+  public TaskResult execute(@Nonnull StageExecution stage) {
+    String cloudProvider = getCloudProvider(stage);
+    String account = stage.mapTo("/service.account", String.class);
+    String region = stage.mapTo("/service.region", String.class);
+    String serviceInstanceName = stage.mapTo("/service.instance.name", String.class);
+
+    Map<String, Object> serviceInstance =
+        oortService.getServiceInstance(account, cloudProvider, region, serviceInstanceName);
+
+    ExecutionStatus status = oortStatusToTaskStatus(serviceInstance);
+
+    TaskResult.TaskResultBuilder taskResultBuilder = TaskResult.builder(status);
+    Optional.ofNullable(serviceInstance)
+        .ifPresent(
+            (si) -> {
+              String lastOperationDescription =
+                  Optional.ofNullable(serviceInstance.get("lastOperationDescription"))
+                      .orElse("")
```

Comment:
Would it make sense to have a generic failure message if the last operation description is null?

Reply:
Yes, makes sense. Added a default one.